pkg/workflow_execution.go

```diff
@@ -691,3 +691,34 @@ func (c *Client) TerminateWorkflowExecution(namespace, name string) (err error)
 
 	return
 }
+
+func (c *Client) GetArtifactDownload(namespace, name, key string) (data []byte, err error) {
+	config, err := c.getNamespaceConfig(namespace)
+	if err != nil {
+		return
+	}
+
+	s3Client, err := c.getS3Client(namespace, config)
```

Comment:
@Vafilor `getS3Client` and `getNamespaceConfig` are now public. Might want to rebase master and test the changes.